app/controllers/users_controller.rb

```diff
@@ -0,0 +1,38 @@
+class UsersController < ApplicationController
+  def index
+    @users = User.all
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      flash[:notice] = 'user was created successfully.'
+      redirect_to root_path
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @user = User.find(params[:id])
+    # @user.update(user_params)
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if @user
+      @user.update(user_params)
+      redirect_to root_path
```

Comment:
**Optional**
it should `redirect_to edit_user_path(<USER_ID>)`. You can change USER_ID to whatever code worked for you.